net/mwan3/files/lib/mwan3/mwan3.sh

```diff
@@ -527,7 +527,10 @@ mwan3_create_iface_rules()
 
 	mwan3_delete_iface_rules "$1"
 
+	network_get_ipaddr wan_addr "${1}"
```

Comment:
Sorry for the late review!
I have some comments about this change that we should think about
* We have to call different functions for [IPv4](https://github.com/openwrt/openwrt/blob/31e45f62cafbc9361c72d320241e589b31e2b5ed/package/base-files/files/lib/functions/network.sh#L30) and [IPv6](https://github.com/openwrt/openwrt/blob/31e45f62cafbc9361c72d320241e589b31e2b5ed/package/base-files/files/lib/functions/network.sh#L37).
* Can we always be sure that we will get an IP form the function `network_get_ipaddr` or network_get_ipaddr6` ?

Reply:
What about the following?
```patch
--- mwan3.sh	2024-09-23 12:34:46.000000000 +0000
+++ /lib/mwan3/mwan3.sh	2024-12-23 18:09:25.564820431 +0000
@@ -519,15 +519,16 @@
 
 	if [ "$family" = "ipv4" ]; then
 		IP="$IP4"
+		network_get_ipaddr wan_addr "${1}"
 	elif [ "$family" = "ipv6" ] && [ $NO_IPV6 -eq 0 ]; then
 		IP="$IP6"
+		network_get_subnet6 wan_addr "${1}"
 	else
 		return
 	fi
 
 	mwan3_delete_iface_rules "$1"
 
-	network_get_ipaddr wan_addr "${1}"
 
 	$IP rule add pref $((id+1000)) iif "$2" lookup "$id"
 	$IP rule add pref $((id+1500)) from ${wan_addr} lookup "$id"
```

Reply:
`network_get_subnet6` may fail, we could skip the rule if it returns 1.